server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -46,12 +47,35 @@ package object server {
       service => service.compose(AuthedRequest(authUser.run))
     }
 
-    /** TODO fs2 port -- replace |||
-    def apply[Err, T](authUser: Service[Request, Err \/ T], onFailure: Kleisli[Task, AuthedRequest[Err], MaybeResponse]): AuthMiddleware[T] = { service =>
-      (onFailure ||| service)
-        .local({authed: AuthedRequest[Either[Err, T]] => authed.authInfo.bimap(err => AuthedRequest(err, authed.req), suc => AuthedRequest(suc, authed.req))})
-        .compose(AuthedRequest(authUser.run))
+    implicit def serviceChoice: Choice[Service] = new Choice[Service] {
```

Comment:
I think this belongs on the `Service` companion object, since it's an instance for `Service`.

Better yet, is this something that could be contributed to cats?  Scalaz provides a choice instance for its Kleisli.  Why can't/doesn't cats?